gemini/getting-started/intro_gemini_2_0_flash_thinking_mode.ipynb

```diff
@@ -0,0 +1,579 @@
+        "print(response.candidates[0].content.parts[0].thought)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "OgaDQSL2C9nn"
+      },
+      "outputs": [],
+      "source": [
+        "Markdown(response.candidates[0].content.parts[0].text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ig6POLZnC9no"
+      },
+      "source": [
+        "Most of the time you won't need to check the thoughts as you'll be mostly interested in the answer, but having access to them gives you a way to check where the answers come from and how the model comes up with it. It's not a black box anymore!\n",
+        "\n",
+        "Then the second part is the actual answer:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7cbb7f5d9faf"
+      },
+      "outputs": [],
+      "source": [
+        "print(response.candidates[0].content.parts[1].thought)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "BJ2aYsAhC9no"
+      },
+      "outputs": [],
+      "source": [
+        "Markdown(response.candidates[0].content.parts[1].text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "3ZqndSMtC9no"
+      },
+      "source": [
+        "As a comparison here's what you'd get with the \"classic\" [Gemini 2.0 Flash](https://cloud.google.com/vertex-ai/generative-ai/docs/gemini-v2) model.\n",
+        "\n",
+        "Unlike thinking mode, the normal model does not articulate its thoughts and tries to answer right away which can lead to more simpler answers to complex problems."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "pY1colD0C9no"
+      },
+      "outputs": [],
+      "source": [
+        "response = client.models.generate_content(\n",
+        "    model=\"gemini-2.0-flash-exp\",\n",
+        "    contents=\"How can I simplify this? `(Math.round(radius/pixelsPerMile * 10) / 10).toFixed(1);`\",\n",
+        ")\n",
+        "\n",
+        "Markdown(response.text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "JmxrcZPfC9no"
+      },
+      "source": [
+        "### **Example 2**: Geometry problem (with image)\n",
+        "\n",
+        "This geometry problem requires complex reasoning and is also using Gemini multimodal capabilities to read the image."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "8ksBfYPZC9no"
+      },
+      "outputs": [],
+      "source": [
+        "!wget https://storage.googleapis.com/generativeai-downloads/images/geometry.png -O geometry.png -q\n",
+        "\n",
+        "im = Image.open(\"geometry.png\").resize((256, 256))\n",
+        "im"
```

Comment:
You don't need to download this image, you can send the uri directly using `Part.from_uri()` And it can be displayed using `IPython.display.Image(url)`